3-projects/modules/single_project/main.tf

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  host_network = data.google_compute_network.shared_vpc
+}
+
+module "project" {
+  source                      = "terraform-google-modules/project-factory/google"
+  version                     = "~> 7.0"
+  random_project_id           = "true"
+  impersonate_service_account = var.impersonate_service_account
+  default_service_account     = "depriviledge"
+  activate_apis               = var.activate_apis
+  name                        = "${var.project_prefix}-${var.environment}"
+  org_id                      = var.org_id
+  billing_account             = var.billing_account
+  folder_id                   = var.folder_id
+
+  shared_vpc         = local.host_network.project
+  shared_vpc_subnets = local.host_network.subnetworks_self_links
+
+  labels = {
+    environment      = var.environment
+    cost_centre      = var.cost_centre
+    application_name = var.application_name
+  }
+}
+
+/******************************************
+  Project subnets
+ *****************************************/
+module "networking_project" {
```

Comment:
Can we comment this out and add (optional) to the comment? @morgante and I discussed this and felt that we should not inflict this opinion on new users, they can decide whether central subnet creation or subnet creation per application works for them

Reply:
Would the enable_network default to false also skip the subnet creation?

Reply:
Yeah I see what you mean, although I think it would be good to make to explicitly something you choose to do rather than a boolean. What I think would be good is:

- Default to centrally managing subnets in the networks pipeline, with project pipeline connecting to those shared vpcs/subnets
- Leave subnet submodule and add documented steps to enable subnet per project in the README, for users who want / need that option.
- Leave private dns submodule and documented steps to enable the private DNS with peering in README

This would leave the basic case nice and simple, but still offer the ability to use those modules where users want. What do you think?